src/libraries/System.Net.HttpListener/src/System/Net/Managed/HttpConnection.cs

```diff
@@ -43,33 +44,33 @@ internal sealed class HttpConnection
     {
         private static AsyncCallback s_onreadCallback = new AsyncCallback(OnRead);
         private const int BufferSize = 8192;
-        private Socket _socket;
+        private Socket? _socket;
         private Stream _stream;
         private HttpEndPointListener _epl;
-        private MemoryStream _memoryStream;
-        private byte[] _buffer;
+        private MemoryStream? _memoryStream;
+        private byte[]? _buffer;
         private HttpListenerContext _context;
-        private StringBuilder _currentLine;
-        private ListenerPrefix _prefix;
-        private HttpRequestStream _requestStream;
-        private HttpResponseStream _responseStream;
+        private StringBuilder? _currentLine;
+        private ListenerPrefix? _prefix;
+        private HttpRequestStream? _requestStream;
+        private HttpResponseStream? _responseStream;
         private bool _chunked;
         private int _reuses;
         private bool _contextBound;
         private bool _secure;
-        private X509Certificate _cert;
+        private X509Certificate? _cert;
         private int _timeout = 90000; // 90k ms for first request, 15k ms from then on
         private Timer _timer;
-        private IPEndPoint _localEndPoint;
-        private HttpListener _lastListener;
-        private int[] _clientCertErrors;
-        private X509Certificate2 _clientCert;
-        private SslStream _sslStream;
+        private IPEndPoint? _localEndPoint;
+        private HttpListener? _lastListener;
+        private int[]? _clientCertErrors;
+        private X509Certificate2? _clientCert;
+        private SslStream? _sslStream;
         private InputState _inputState = InputState.RequestLine;
         private LineState _lineState = LineState.None;
         private int _position;
 
-        public HttpConnection(Socket sock, HttpEndPointListener epl, bool secure, X509Certificate cert)
+        public HttpConnection(Socket sock, HttpEndPointListener epl, bool secure, X509Certificate? cert)
```

Comment:
`cert` argument is nullable there but later the nullability is override by `!`. Is it necessary to have it nullable here?

Reply:
Ah, yea you're right. `_cert` need not be nullable. Fixed!